src/puppetlabs/puppetdb/command.clj

```diff
@@ -131,27 +132,29 @@
           (string? mq-endpoint)
           (string? raw-command)]
    :post [(string? %)]}
+  ;; FIXME: is this efficient enough?  It's rebuilding the whole MQ
```

Comment:
This is what we talked about before, right? I'd love to see this fixed, but I'd be totally fine with making it a separate ticket.o